cf-agent/verify_processes.c

```diff
@@ -231,12 +233,29 @@ static PromiseResult VerifyProcessOp(EvalContext *ctx, const Attributes *a, cons
             {
                 if (IsExecutable(CommandArg0(a->process_stop)))
                 {
-                    ShellCommandReturnsZero(a->process_stop, SHELL_TYPE_NONE);
+                    Log(LOG_LEVEL_DEBUG, "Found process_stop command '%s' is executable.", a->process_stop);
+
+                  if (ShellCommandReturnsZero(a->process_stop, SHELL_TYPE_NONE))
+                  {
+                      cfPS(ctx, LOG_LEVEL_INFO, PROMISE_RESULT_CHANGE, pp, a,
+                           "Promise to stop '%s' repaired, '%s' returned zero",
+                           pp->promiser, a->process_stop);
+                      result = PromiseResultUpdate(result, PROMISE_RESULT_CHANGE);
+                      return result;
+                  }
+                  else
+                  {
+                    cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, a,
+                         "Promise to stop '%s' failed, '%s' returned nonzero",
+                         pp->promiser, a->process_stop);
+                    result = PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
+                    return result;
```

Comment:
In this case, it might be appropriate to return early, like you do, since it's erroring. You need to `DeleteItemList(killlist)` first, though.

Reply:
Right, that was my reasoning for the guess.

Reply:
Added